src/Command/EnumCommand.php

```diff
@@ -51,4 +55,45 @@ public function template(): string
     {
         return 'Bake.Model/enum';
     }
+
+    /**
+     * Get template data.
+     *
+     * @param \Cake\Console\Arguments $arguments The arguments for the command
+     * @return array
+     * @phpstan-return array<string, mixed>
+     */
+    public function templateData(Arguments $arguments): array
+    {
+        $data = parent::templateData($arguments);
+
+        $backed = $arguments->getOption('backed');
+        if ($backed && !in_array($backed, ['string', 'int'], true)) {
+            throw new InvalidArgumentException('Backed enums must be of type `string` or `int`');
+        }
+
+        $data['backed'] = $backed;
+
+        return $data;
+    }
+
+    /**
+     * Gets the option parser instance and configures it.
+     *
+     * @param \Cake\Console\ConsoleOptionParser $parser The option parser to update.
+     * @return \Cake\Console\ConsoleOptionParser
+     */
+    public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        $parser = $this->_setCommonOptions($parser);
+
+        $parser->setDescription(
+            'Bake (backed) enums for use in models.'
+        )->addOption('backed', [
+            'help' => 'If using backed enums. Set to `string` or `int`.',
```

Comment:
For the db at least you can only use backed enums. So we can skip the "If using...".

Reply:
Also since there are only 2 valid values we can use the `choices` config as done [here](https://github.com/cakephp/bake/blob/3.x/src/Command/TestCommand.php#L717) with default as string.

Reply:
That only works for arguments, not options though

Reply:
Rephrased the help sentence.

Reply:
>  That only works for arguments, not options though

Options also have choices https://github.com/cakephp/cakephp/blob/5.x/src/Console/ConsoleInputOption.php#L101

Reply:
If anyone has time  feel free to add on too
I am now offline for a bit.